File: dcp-solutions/2024-12-02/solution.ex
```elixir
defmodule TreeNode {
  @enforce_keys [:val, :left, :right]
  defstruct val: nil, left: nil, right: nil
}

defmodule BinaryTreeInverter {
  @spec invert_tree(TreeNode.t()) :: TreeNode.t()
  defstruct(_)

  def invert_tree(nil) do
    nil
  end

  def invert_tree(tree) when is_map() do
    %TreeNode{val: tree["val"], left: invert_tree(map_get(tree, :left)), right: invert_tree(map_get(tree, :right))} |> Map.new()
  end

  def invert_tree(tree) when is_list() do
    %TreeNode{
      val: tree[0]["val"],
      left: invert_tree(Enum.map(&1, &2["left"])),
      right: invert_tree(Enum.map(&1, &2["right"]))
    } |> Map.new()
  end
}
```

File: dcp-solutions/2024-12-02/solution.php
```php
class TreeNode {
    public $val = null;
    public $left = null;
    public $right = null;

    function __construct($val = 0, $left = null, $right = null) {
        $this->val = $val;
        $this->left = $left;
        $this->right = $right;
    }
}

function invertTree($root) {
    if ($root == null) {
        return null;
    }

    $temp = $root->left;
    $root->left = $root->right;
    $root->right = $temp;

    invertTree($root->left);
    invertTree($root->right);

    return $root;
}
```